natsort_test.go

```diff
@@ -67,6 +72,22 @@ func less(a, b string) bool {
 	return a < b
 }
 
+var dre = regexp.MustCompile(`\d+`)
+
+func pad(str string) string {
+	f := func(match []byte) []byte {
+		n, _ := strconv.ParseInt(string(match), 10, 64)
```

Comment:
This code is inefficient. Given that you explicitly only look for decimal, you can just do something like:

```
if n := 10 - len(match); n > 0 {
  x := [10]byte{}
  copy(x[n:], match)
  return x[:]
}
return match
```

Reply:
This code fails with

```
--- FAIL: BenchmarkLessDigitsTwoGroupsDummy
    bench_test.go:136: unexpected result
```

And existing one did work for quite a long in production until https://github.com/go-graphite/carbonapi/pull/673, and it's indeed ineffective